File: leet/0001-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    if nums == nil || len(nums) == 0 {
        return nil
    }
    
    var numMap = map[int]int{}
    for i, val := range nums {
        n, ok := numMap[ target - val ]
        if(ok){
            return []int{n, i}
        }else{
            numMap[val] = i
        }
    }
    
    return nil
}

File: leet/0001-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        rs = []
        dist = {}
        
        for i in range(len(nums)):
            if target - nums[i] in dist.keys():
                rs.append(i)
                rs.append(dist[target - nums[i]])
                return rs
            else :
                dist[nums[i]] = i
        
        return rs
